ReactiveCocoa/Objective-C/RACSignal.m

```diff
@@ -107,31 +107,36 @@ - (RACSignal *)bind:(RACStreamBindBlock (^)(void))block {
 	return [[RACSignal createSignal:^(id<RACSubscriber> subscriber) {
 		RACStreamBindBlock bindingBlock = block();
 
-		NSMutableArray *signals = [NSMutableArray arrayWithObject:self];
+		__block NSUInteger signalsCount = 1;   // indicates self
```

Comment:
Since `-bind:` is super popular, it will probably be more efficient to use atomic operations instead of a lock:
```
__block volatile int32_t signalsCount = 1;

// Then call:
OSAtomicIncrement32Barrier(&signalsCount);
OSAtomicDecrement32Barrier(&signalsCount);
```